docs/migration-guide/w3c-vc-v2.md

```diff
@@ -0,0 +1,565 @@
+---
+
+# Migration Guide: TrustVC W3C VC Data Model v1.1 to v2.0
+
+This guide provides comprehensive instructions for migrating **TrustVC library** implementations from **W3C VC Data Model v1.1** to **v2.0**, covering both technical implementation details for developers and practical usage examples for users.
+
+> **Note**: This migration guide is specifically designed for users of the **TrustVC library** (`@trustvc/trustvc` packages). If you're using other W3C VC implementations, please refer to their respective documentation.
+
+## Overview
+
+The TrustVC library's support for W3C VC Data Model v2.0 introduces significant improvements over v1.1, including:
+
+- **Modern Cryptographic Suites**: ECDSA-SD-2023 with selective disclosure capabilities
+- **Updated Context URLs**: New v2.0 contexts for enhanced interoperability
+- **Improved Date Fields**: `validFrom`/`validUntil` replacing `issuanceDate`/`expirationDate`
+- **Enhanced Status Lists**: BitstringStatusList replacing StatusList2021
+- **Multikey Format**: Modern key representation for better DID document compatibility
+
+## Key Differences Summary
+
+| Aspect | v1.1 (Legacy) | v2.0 (Modern) |
+|--------|---------------|---------------|
+| **Context** | `https://www.w3.org/2018/credentials/v1` | `https://www.w3.org/ns/credentials/v2` |
+| **Date Fields** | `issuanceDate`, `expirationDate` | `validFrom`, `validUntil` |
+| **Cryptosuite** | BBS+ (`BbsBlsSignature2020`) | ECDSA-SD-2023 (`ecdsa-sd-2023`) |
+| **Proof Type** | `BbsBlsSignature2020` | `DataIntegrityProof` |
+| **Key Format** | BLS12-381 G2 (`privateKeyBase58`) | Multikey (`secretKeyMultibase`) |
+| **Status List** | `StatusList2021Credential` | `BitstringStatusListCredential` |
+| **Status Entry** | `StatusList2021Entry` | `BitstringStatusListEntry` |
+| **Status Subject** | `StatusList2021` | `BitstringStatusList` |
+
+## Migration Steps
+
+### 1. Update Dependencies
+
+First, ensure you're using the latest versions of TrustVC packages that support v2.0:
+
+```bash
+npm install @trustvc/trustvc@latest
+```
+
+### 2. Key Pair Migration
+
+#### v1.1 Key Generation (Legacy)
+```typescript
+import { generateKeyPair, VerificationType } from '@trustvc/w3c-issuer';
+
+// v1.1 - BLS12-381 G2 Key
+const legacyKeyPair = await generateKeyPair({
+  type: VerificationType.Bls12381G2Key2020,
+  seedBase58: 'your-seed-here' // optional
+});
+
+console.log(legacyKeyPair);
+// Output:
+// {
+//   type: "Bls12381G2Key2020",
+//   privateKeyBase58: "...",
+//   publicKeyBase58: "...",
+//   seedBase58: "..."
+// }
+```
+
+#### v2.0 Key Generation (Modern)
+```typescript
+import { generateKeyPair, CryptoSuite } from '@trustvc/w3c-issuer';
+
+// v2.0 - ECDSA-SD-2023 with Multikey
+const modernKeyPair = await generateKeyPair({
+  type: CryptoSuite.EcdsaSd2023 // 'ecdsa-sd-2023'
+});
+
+console.log(modernKeyPair);
+// Output:
+// {
+//   type: "Multikey",
+//   secretKeyMultibase: "z...",
+//   publicKeyMultibase: "z..."
+// }
+```
+
```

Comment:
Can add a note to remind users to update their DID:WEB with the new key? show a sample of the DID doc. manual update way can do.